metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -681,30 +691,16 @@ def form_path(project_id: str) -> List[str]:
 
     def _is_allowed_project(self, project: TableauProject) -> bool:
         # Either project name or project path should exist in allow
-        is_allowed: bool = self.config.project_pattern.allowed(
```

Comment:
We encountered some issues with the current implementation. The problem is that both project paths and project names can be controlled with the same pattern.

For example, if you deny a nested project by its path e.g. "Project1/Nested1/Test (Catalog)" but allow all projects with "(Catalog)" in the name (by a regex). The denied project "Project1/Nested1/Test (Catalog)" would still be ingested as the project name is allowed. Or if we, for example want to deny all projects with "(Sensitive)" in the **path** this can't be achieved because the names might be allowed.

Ideally, I think it would make sense to completely separate `project_path_pattern` and `project_name_pattern` to avoid these cases. But this would break backwards compatibility.

What do you think about this?